test/e2e/on-request-error/server-action-error/server-action-error.test.ts

```diff
@@ -0,0 +1,110 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry } from 'next-test-utils'
+
+describe('on-request-error - server-action-error', () => {
+  const { next, skipped } = nextTestSetup({
+    files: __dirname,
+    skipDeployment: true,
+    env: {
+      __NEXT_EXPERIMENTAL_INSTRUMENTATION: '1',
+    },
+  })
+
+  if (skipped) {
+    return
+  }
+
+  const outputLogPath = 'output-log.json'
+
+  async function getOutputLogJson() {
+    if (!(await next.hasFile(outputLogPath))) {
+      return {}
+    }
+    const content = await next.readFile(outputLogPath)
+    return JSON.parse(content)
+  }
+
+  async function validateErrorRecord({
+    errorMessage,
+    url,
+  }: {
+    errorMessage: string
+    url: string
+  }) {
+    // Assert the instrumentation is called
+    await retry(async () => {
+      const recordLogs = next.cliOutput
+        .split('\n')
+        .filter((log) => log.includes('[instrumentation] write-log'))
+      const expectedLog = recordLogs.find((log) => log.includes(errorMessage))
+      expect(expectedLog).toBeDefined()
+    }, 5000)
+
+    const json = await getOutputLogJson()
+    const record = json[errorMessage]
+
+    // Assert error is recorded in the output log
+    expect(record).toMatchObject({
+      payload: {
+        message: errorMessage,
+        request: {
+          url,
+          method: 'POST',
+          headers: expect.objectContaining({
+            accept: 'text/x-component',
+          }),
+        },
+        context: {
+          routerKind: 'App Router',
+          routeType: 'action',
+          renderSource: 'react-server-components-payload',
+        },
+      },
+      count: 1,
+    })
+  }
+
+  beforeAll(async () => {
```

Comment:
Shouldn't that be `beforeEach`?

Reply:
It's for reseting the records before recording them again, the key is the errorMessage, so they all share the same object